sotodlib/tod_ops/apodize.py

```diff
@@ -1,24 +1,86 @@
 import numpy as np
 
+def get_apodize_window_for_ends(aman, apodize_samps=1600):
+    """
+    Generate an apodization window using a cosine taper at the beginning and end.
+
+    Args:
+        aman: An axismanager
+        apodize_samps (int): Number of samples to apply the cosine taper to at each end.
+
+    Returns:
+        numpy.ndarray: An array representing the apodization window.
+    """
+    w = np.ones(aman.samps.count)
+    cosedge = np.cos(np.linspace(0, np.pi/2, apodize_samps))
+    w[-apodize_samps:] = cosedge
+    w[:apodize_samps] = np.flip(cosedge)
+    return w
+    
+def get_apodize_window_from_flags(aman, flags, apodize_samps=200):
+    """
+    Generate an apodization window based on flag values.
```

Comment:
I think you want to elaborate on this doctoring a bit more. You are essentially cosine tapering every continuous portion of data between flagged regions, make that explicit in the docstring